### DISCUSSION TITLE
PerceptualHash is quite different on version 14.0.0 vs prior versions. Expected?

### DISCUSSION THREAD

**Discussion by ozziepeeps** [dlemstra/Magick.NET#1720] [Help] [answered]
Hi, thanks for a great library!

I have observed that the `PerceptualHash` generated by version `14.0.0` is quite different to prior versions. I have created a repo that demonstrates what I'm seeing: https://github.com/ozziepeeps/MagickNET-perceptualhash-change

The repo includes a single (public domain) image asset. I have then written some unit tests and generated the perceptual hash on:

- `Magick.NET-Q16-AnyCPU 7.16.0`
- `Magick.NET-Q16-AnyCPU 13.10.0`
- `Magick.NET-Q16-AnyCPU 14.0.0`

All 3 package versions are present in the csproj for your convenience.

I observe that the PerceptualHash changes ever so slightly between 7.16.0 and 13.10.0, but the unit test `HashDifference_MagickNET_Q16_AnyCPU_7_16_0_VS_MagickNET_Q16_AnyCPU_13_10_0` demonstrates that the difference is negligible `< 0.00001`.

However the unit test `HashDifference_MagickNET_Q16_AnyCPU_13_10_0_VS_MagickNET_Q16_AnyCPU_14_0_0` demonstrates that the difference between 13.10.0 and 14.0.0 is much larger: `24.356`.

I am not sure if this is intended. I've observed pretty stable / negligible changes across versions between 7.16.0 and 13.10.0 over my years of using the NuGet, but this one stood out so I wanted to highlight it just in case it wasn't intended.

Thanks!

**Comment by dlemstra** [comment, marked as answer]
Thanks for reaching out but this was an intended change and mentioned in the changelog of release 14.0.0: https://github.com/dlemstra/Magick.NET/releases/tag/14.0.0

- Changed the default colorspaces for PerceptualHash from sRGB and HCLp to XyY and HSB

You will get the "old" result if you do `image.PerceptualHash(ColorSpace.sRGB, ColorSpace.HCLp)` instead. This was an intended change because ImageMagick changed it's default implementation a while back. And with a major release I could finally make that change in this library.

**Reply by ozziepeeps** [reply]
Ah! I missed that when going through the release notes. My bad 👍 I will try your workaround for the "old" result in a few hours and report back here. 

Whilst I have your attention, what is the theoretical maximum "difference" one can get from `SumSquaredDistance`? I don't recall ever being aware of or seeing a maximum value...

**Reply by dlemstra** [reply]
I don't know what the maximum value can be. I don't have the math expertise to give you an answer for that question.

**Reply by ozziepeeps** [reply]
No problem! I can confirm that the workaround you suggested works for me. I shall mark your reply as the answer. Thanks again!
